data_structures/arrays/permutations.py

```diff
@@ -0,0 +1,31 @@
+from itertools import permutations
+
+
+# All permutations:
+def permute(nums: list[int]) -> list[list[int]]:
+    """
+    >>> from itertools import permutations
+    >>> numbers= [1,2,3]
+    >>> all(list(nums) in permute(numbers) for nums in permutations(numbers))
```

Comment:
Please add more tests like when the input is an empty list, list with 1 number, multiple numbers, etc. Test out the algorithm as a [black box](https://en.wikipedia.org/wiki/Black-box_testing) and [white box](https://en.wikipedia.org/wiki/White-box_testing).